vsml_core/src/schemas.rs

```diff
@@ -117,6 +119,297 @@ impl FromStr for Order {
+    "snow" => FontColor(255, 250, 250, 255),
+    "springgreen" => FontColor(0, 255, 127, 255),
+    "steelblue" => FontColor(70, 130, 180, 255),
+    "tan" => FontColor(210, 180, 140, 255),
+    "teal" => FontColor(0, 128, 128, 255),
+    "thistle" => FontColor(216, 191, 216, 255),
+    "tomato" => FontColor(255, 99, 71, 255),
+    "turquoise" => FontColor(64, 224, 208, 255),
+    "violet" => FontColor(238, 130, 238, 255),
+    "wheat" => FontColor(245, 222, 179, 255),
+    "white" => FontColor(255, 255, 255, 255),
+    "whitesmoke" => FontColor(245, 245, 245, 255),
+    "yellow" => FontColor(255, 255, 0, 255),
+    "yellowgreen" => FontColor(154, 205, 50, 255),
+};
+
+impl FromStr for FontColor {
+    type Err = FontColorParseError;
+
+    fn from_str(value: &str) -> Result<FontColor, Self::Err> {
+        match value {
+            v if v.starts_with('#') => {
+                let Some(hex) = v.strip_prefix('#') else {
+                    return Err(FontColorParseError::UnknownMode);
+                };
+                let Ok(num_value) = u32::from_str_radix(hex, 16) else {
+                    return Err(FontColorParseError::UnknownMode);
+                };
+                match hex.len() {
+                    3 => {
+                        let r = (((num_value >> 8) & 0xF) * 0x11) as u8;
+                        let g = (((num_value >> 4) & 0xF) * 0x11) as u8;
+                        let b = ((num_value & 0xF) * 0x11) as u8;
+                        let a = 255;
+                        Ok(FontColor(r, g, b, a))
+                    }
+                    4 => {
+                        let r = (((num_value >> 12) & 0xF) * 0x11) as u8;
+                        let g = (((num_value >> 8) & 0xF) * 0x11) as u8;
+                        let b = (((num_value >> 4) & 0xF) * 0x11) as u8;
+                        let a = ((num_value & 0xF) * 0x11) as u8;
+                        Ok(FontColor(r, g, b, a))
+                    }
+                    6 => {
+                        let r = ((num_value >> 16) & 0xFF) as u8;
+                        let g = ((num_value >> 8) & 0xFF) as u8;
+                        let b = (num_value & 0xFF) as u8;
+                        let a = 255;
+                        Ok(FontColor(r, g, b, a))
+                    }
+                    8 => {
+                        let r = ((num_value >> 24) & 0xFF) as u8;
+                        let g = ((num_value >> 16) & 0xFF) as u8;
+                        let b = ((num_value >> 8) & 0xFF) as u8;
+                        let a = (num_value & 0xFF) as u8;
+                        Ok(FontColor(r, g, b, a))
+                    }
+                    _ => Err(FontColorParseError::UnknownMode),
+                }
+            }
+            v if v.starts_with("rgb(") => {
+                let rgb_regex = Regex::new(r"^rgb\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\)$")
+                    .map_err(|_| FontColorParseError::UnknownMode)?;
```

Comment:
Regex::newは若干重めなので一回構築したものを使いまわすようにしましょう
あとここのエラー処理はunwrapでよいです(なぜなら文字列リテラルの内容によってのみ発生するエラーで回復を想定しないので)
```suggestion
                static RGB_REGEX: LazyLock<Regex> = LazyLock::new(|| Regex::new(r"^rgb\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\)$").unwrap());
```